project/threads/models.py

```diff
@@ -102,8 +102,8 @@ def image_url(self):  # TODO: move this to utils
     is_draft = models.BooleanField(default=True)
 
     num_views = models.IntegerField(default=0)
-    num_civis = models.IntegerField(default=0)
-    num_solutions = models.IntegerField(default=0)
+    num_civis = models.IntegerField(default=0)  # TODO To be dropped
+    num_solutions = models.IntegerField(default=0)  # TODO To be dropped
```

Comment:
Do we still intend to show these counts, such as on the Thread page? If so, we should consider *when* / *how often* this count should run. E.g., should the count be a cached property on the Thread model that is only updated when Civis/Solutions are created/destroyed? Or do we want to run the `count` queries *every* time the template is rendered?

Reply:
We shouldn't store number of civis and solutions to the database because we can directly get these stats with:

```python
num_civis = Thread.objects.get(id=thread_id).civis.count()
num_solutions = Thread.objects.get(id=thread_id).civis.solutions.count()
```

Since the aim of this task is to port thread template to Django template syntax I didn't want to do that in this PR. That is the reason why I added `# TODO To be dropped` comments to these fields, just for future reference. If I dropped them in this PR, then I have to attach makemigrations file.

These numbers show the number of civis and solutions in only one thread, not all threads. So we don't need cached property at this point. We want `count` queries whenever a user hit the thread-detail page and the corresponding template is rendered. 

Right now adding and removing civi functionalities don't work as expected. While fixing them, we. should take into account that when a civi is created or removed we should update number of civis with JavaScript. Good catch. 👍

Reply:
Performance won't be a problem when we need count of objects. It is not a huge burden on our backend. If you have some concerns about the performance we can count the number of civis in frontend. 😄  These are all early optimizations but as I said they don't affect the response time. When I work on porting account and thread templates to Django template syntax, I get an insight that we should reduce the number of lines in the project to make the project maintainable, which will make it easier for new contributors to involve in the project.

Reply:
We can also drop these fields:

```
    votes_vneg = models.IntegerField(default=0)
    votes_neg = models.IntegerField(default=0)
    votes_neutral = models.IntegerField(default=0)
    votes_pos = models.IntegerField(default=0)
    votes_vpos = models.IntegerField(default=0)
```